src/main/java/pl/javastart/task/services/BillPayPhoneContract.java

```diff
@@ -0,0 +1,41 @@
+package pl.javastart.task.services;
+
+public class BillPayPhoneContract extends Contract {
+    private int billPayFee;
+
+    public BillPayPhoneContract(int billPayFee) {
+        super.setCreditBalance(billPayFee);
+        this.billPayFee = billPayFee;
+    }
+
+    public int getBillPayFee() {
+        return billPayFee;
+    }
+
+    protected void setBillPayFee(int billPayFee) {
+        this.billPayFee = billPayFee;
+    }
+
+    @Override
+    protected boolean checkCallAvailability(int seconds) {
+        setVoiceEventsTime(getVoiceEventsTime() + seconds);
+        return true;
+    }
+
+    @Override
+    protected boolean checkSmsAvailability() {
+        setSmsAmount(getSmsAmount() + 1);
```

Comment:
Możesz to zastąpić: `smsAmount++` gdy ustawisz widoczność `smsAmount` na `protected` zamiast private. Analogicznie w przypadku mmsów i sekund rozmowy.